dev/core/src/com/google/gwt/dev/cfg/ModuleDef.java

```diff
@@ -455,31 +455,24 @@ public String getGwtXmlFilePath(String moduleName) {
    * For example, consider a glob that matches fewer files than before because a file was
    * deleted.
    */
-  public int getInputFilenameHash() {
-    List<String> filenames = new ArrayList<String>();
-
-    filenames.addAll(gwtXmlPathByModuleName.values());
+  public long getInputFilenameHash() {
+    List<String> filenames = new ArrayList<>(gwtXmlPathByModuleName.values());
 
     for (Resource resource : getResourcesNewerThan(Integer.MIN_VALUE)) {
       filenames.add(resource.getLocation());
     }
 
-    // Take the first four bytes of the SHA-1 hash.
+    // Take the first eight bytes of the murmur3 hash.
```

Comment:
There's no reason that this method ever needed to return 4 bytes to begin with (especially of a sha1, which is 20 bytes wide). Even 8 bytes is a little silly of a 16 byte murmur, since the Recompiler.InputSummary could just as easily hold a string of the hash. The InputSummary only exists until the next recompile to quickly check if the inputs are the same or different.

I'm going to go ahead and change this all to use the full string. I don't understand how this code came about to begin with.